homework/App.js

```diff
@@ -17,13 +17,42 @@ class App {
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { text: 'HYF Repositories', class: 'hyf' });
+    const selectMenu = Util.createAndAppend('select', header, { class: 'repository-selector' });
+    const containerDiv = Util.createAndAppend('div', root, { id: 'container' });
+    const repoDiv = Util.createAndAppend('div', containerDiv, { class: 'left-div' });
+    const table = Util.createAndAppend('table', repoDiv);
+
+    const contributorsDiv = Util.createAndAppend('div', containerDiv, { class: 'right-div' });
+    Util.createAndAppend('p', contributorsDiv, {
+      text: 'Contributors',
+      class: 'contributor-header',
+    });
+    const contributorList = Util.createAndAppend('ul', contributorsDiv, {
+      class: 'contributor-list',
+      id: 'contributorListId',
+    });
 
     try {
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos = repos
+        .sort((a, b) => a.name.localeCompare(b.name, 'en', { sensivity: 'base' }))
+        .map(repo => new Repository(repo));
+
+      repos.forEach((repo, i) => {
+        Util.createAndAppend('option', selectMenu, { text: repo.name, value: i });
+      });
+      const newRepo = new Repository(repos);
+      newRepo.render(table, 0);
+      this.fetchContributorsAndRender(0);
+      selectMenu.addEventListener('change', event => {
+        App.clearContainer(table);
+        const index = event.target.value;
+        newRepo.render(table, index);
```

Comment:
we said about the index isn't important any more for the repository it self, so you may type the following using the sorted repositories you have from line 39.
```suggestion
        this.repos[index].render(table);
```

Note that we are using the index for the repos array by it self not the render method making the render free of any un-needed addition.

We didn't finish yet please follow the next comment.

Reply:
I understand the point about using repo without index parameter.